pkg/hub/client/policy_enforced_client.go

```diff
@@ -298,6 +293,7 @@ func (e *PolicyEnforcedClient) PopWorkflowStepPolicy() {
 // Policy gets policy which the Client uses. This getter is thread safe.
 func (e *PolicyEnforcedClient) Policy() policy.Policy {
 	e.mu.Lock()
+	e.mergePolicies()
```

Comment:
There are two approaches: 
(1) keep the `e.mergedPolicy` in order to have `resolvePolicyTIMetadataIfShould` as a separate function, and call it only when necessary. Thus, we don't need to have to handle the propagation of the hub error in all test cases and so on. 

(2) Remove the ``e.mergedPolicy` and propagate the error - mainly in `ListTypeInstancesBackendsBasedOnPolicy` and all test cases. 

For now, I don't have a strong opinion, and I stick to using (1) approach but I can adjust it if needed quickly as it is a simple change.

Reply:
🤔  I don't understand - `e.mergedPolicy` is not used anywhere apart from the `Policy()` getter, so we can safely get rid of it 🙂 

While propagating error from `mergePolicies` would be great, let's skip it. Here's the commit I introduced such change: https://github.com/pkosiec/capact/commit/7fe8a06a9097afda756b4dd59629c54671afb2cf Please apply the changes from the commit and we will be ready to merge 🙂